js/context.js

```diff
@@ -1,252 +1,241 @@
-
-			targetParent = closest(e.target, "A") || document.createElement('A');
-
-			parentHTML = targetParent.cloneNode(false);
-			parentHTML.href = parentHTML.href;
-
-			targetImage = e.target.cloneNode(false);
-			targetImage.src = targetImage.src;
-
-			parentHTML.appendChild(targetImage);
-			e.dataTransfer.setData("text/html", parentHTML.outerHTML);
-		}
-
-		markletDiv.style.display = "block";
-	}
-
-	function stopDrag(e) {
-		markletDiv.style.display = "none";
-	}
-
-	function dragOver(e) {
-		e.stopPropagation();
-		e.preventDefault();
-		return false;
-	}
-
-	function dragEnter(e) {
-		sendMessage({
-			action: "dragenter"
-		});
-		return false;
-	}
-
-	function dragLeave(e) {
-		sendMessage({
-			action: "dragleave"
-		});
-		return false;
-	}
-
-	function drop(e) {
-		var data = {
-			source: location.href,
-			title: document.title
-		};
-
-		for (var i in e.dataTransfer.types) {
-			data[e.dataTransfer.types[i]] = e.dataTransfer.getData(e.dataTransfer.types[i]);
-		}
+  function createFrame(msg) {
+    markletFrame = document.createElement("iframe");
+    markletFrame.name = markletFrame.id = "arena_frame";
+    if (msg) {
+      markletFrame.src = getURL(msg);
+    } else {
+      markletFrame.src = getURL({
+        url: window.location.href,
+        title: window.document.title
+      });
+    }
+
+    document.body.appendChild(markletFrame);
+  }
+
+  function getURL(msg) {
+    baseUrl = "https://www.are.na/save/";
+
+    if(msg.options.srcUrl) {
+      path = msg.options.srcUrl;
+    } else if(msg.options.selectionText) {
+      path = msg.options.selectionText;
+    } else if(msg.options.linkUrl) {
+      path = msg.options.linkUrl;
+    } else {
+      path = msg.options.pageUrl;
+    }
+
+    url = baseUrl + encodeURIComponent(path);
+    url += "?original_source_url=" + encodeURIComponent(msg.url);
+    url += "&original_source_title=" + encodeURIComponent(msg.title);
+
+    return url;
+  }
+
+  function createTarget() {
+    markletDiv = document.createElement("div");
+    markletDiv.id = "arena_div";
+
+    document.body.appendChild(markletDiv);
+  }
+
+  function closeBookmarklet() {
+    if (markletFrame) document.body.removeChild(markletFrame);
+    if (markletDiv) document.body.removeNode(markletDiv);
+    if (markletStyle) document.body.removeNode(markletStyle);
+  }
+
+  function createStyle() {
+    var url = chrome.extension.getURL("css/style.css");
+    var head = document.head;
+    var link = document.createElement('link');
+
+    link.type = 'text/css';
+    link.rel = 'stylesheet';
+    link.href = url;
+
+    head.appendChild(link);
+  }
+
+  function closest(that, tagname) {
+    while (that.parentNode) {
+      that = that.parentNode;
+      if (that.tagName == tagname) {
+        return that;
+      }
+    }
+    return false;
+  }
+
+  function sendMessage(data) {
+    markletFrame.contentWindow.postMessage(data, "*");
```

Comment:
Not specific to your PR, but I get a null error here when the save request is submitted to are.na. Save action still succeeds though.